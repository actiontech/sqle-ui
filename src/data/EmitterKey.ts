enum EmitterKey {
  Refresh_User_list = 'REFRESH_USER_LIST',
  Refresh_Role_list = 'REFRESH_ROLE_LIST',
  Refresh_Whitelist_List = 'REFRESH_WHITELIST_LIST',

  Refresh_Rule_Template_List = 'REFRESH_RULE_TEMPLATE_LIST',

  Reset_Create_Order_Form = 'RESET_CREATE_ORDER_FORM',

  Reset_Test_Data_Source_Connect = 'RESET_TEST_DATA_SOURCE_CONNECT',

  Reset_Workflow_Template_Form = 'RESET_WORKFLOW_TEMPLATE_FORM',

  Rest_Audit_Plan_Form = 'REST_AUDIT_PLAN_FORM',

  Refresh_Audit_Plan_Record = 'REFRESH_AUDIT_PLAN_RECORD',
}

export default EmitterKey;
