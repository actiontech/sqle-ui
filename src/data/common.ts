export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}