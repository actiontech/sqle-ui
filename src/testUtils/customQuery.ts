import * as domTestingLib from '@testing-library/dom';
import { fireEvent, screen } from '@testing-library/react';

export const getBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  if (temp.length > 1) {
    throw domTestingLib.queryHelpers.getElementError(
      `Found multiple elements with the selector=${selector}`,
      document.body
    );
  }
  return temp[0];
};

export const getAllBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  return temp;
};

export const getSelectContentByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item-content',
    screen.getByText(label).parentNode?.parentNode as HTMLDivElement
  );
};

export const getSelectValueByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item',
    screen.getByLabelText(label).parentNode?.parentNode as HTMLDivElement
  );
};

/**
 * 模拟选择 Select 组件中下拉框中的某项数据
 * @param 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionText 需要选中的下拉项
 * @param index 通过文本获取所有的节点会有两个(具体信息可以在网页上查看Select 组件中下拉的节点信息), 正常来说其中第二个是需要来需要用户点击的,
 * 所以这里的 index 默认值为 1. 并且这个节点拥有类名 ant-select-item-option-content, 所以在点击前需要使用 expect 来判断是否为正确的节点.
 */
export const selectOptionByIndex = (
  label: string,
  optionText: string,
  index = 1
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = screen.getAllByText(optionText);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }
  const option = options[realIndex];
  expect(option).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

/**
 * 具体功能同 selectOptionByIndex, 不过该函数主要针对下拉框选项为一个节点的情况
 * @param label 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionCls 下拉项节点最外层的 class
 * @param index  这里不同于 selectOptionByIndex 的 index, 其代表的就是真实需要选择的下拉项的序号, 从 0 开始.
 */
export const selectCustomOptionByClassName = (
  label: string,
  optionCls: string,
  index: number
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = getAllBySelector(`.${optionCls}`);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }

  const option = options[realIndex];
  expect(option.parentNode).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

export const getHrefByText = (name: string) => {
  return screen.getByText(name).getAttribute('href');
};

export const getAllHrefByText = (name: string) => {
  return screen.getAllByText(name).map((v) => v.getAttribute('href'));
};
