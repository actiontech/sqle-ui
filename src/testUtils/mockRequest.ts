import { AxiosResponse } from 'axios';
import instance from '../api/instance';
import role from '../api/role';
import ruleTemplate from '../api/rule_template';
import user from '../api/user';
import workflow from '../api/workflow';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData,
  };
};

export const failedData = (data?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data),
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data),
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {},
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData),
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const throwErrorThreeSecond = (
  error = 'error',
  { status = 500, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise((res) => {
    throw new Error(error);
  });
};

export const mockUseInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ instance_name: 'instance1' }])
  );
  return spy;
};

export const mockUseInstanceSchema = () => {
  const spy = jest.spyOn(instance, 'getInstanceSchemasV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({ schema_name_list: ['schema1'] })
  );
  return spy;
};

export const mockUseRole = () => {
  const spy = jest.spyOn(role, 'getRoleTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ role_name: 'role_name1' }])
  );
  return spy;
};

export const mockUseRuleTemplate = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ rule_template_name: 'rule_template_name1' }])
  );
  return spy;
};

export const mockUseUsername = () => {
  const spy = jest.spyOn(user, 'getUserTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ user_name: 'user_name1' }])
  );
  return spy;
};

export const mockUseWorkflowTemplate = () => {
  const spy = jest.spyOn(workflow, 'getWorkflowTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ workflow_template_name: 'workflow-template-name-1' }])
  );
  return spy;
};
