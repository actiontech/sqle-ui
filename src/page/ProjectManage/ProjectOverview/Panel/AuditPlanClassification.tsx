import { Result, Space, Typography } from 'antd';
import PanelWrapper from './PanelWrapper';
import { useTranslation } from 'react-i18next';
import { Link } from '../../../../components/Link';
import { PanelCommonProps } from '.';
import { Column, ColumnConfig } from '@ant-design/plots';
import { useMemo, useState } from 'react';
import { projectOverviewData } from '../index.data';
import usePanelCommonRequest from './usePanelCommonRequest';
import statistic from '../../../../api/statistic';

const { rowHeight, thirdLineSize } = projectOverviewData;

const config: ColumnConfig = {
  data: [],
  xField: 'type',
  yField: 'value',
  isStack: true,
  seriesField: 'category',
  color: ['#fbd44d', '#43cb77', '#42a2ff'],
  minColumnWidth: 6,
  maxColumnWidth: 40,
  legend: {
    position: 'bottom',
  },
  tooltip: {
    shared: false,
  },
};

const AuditPlanClassification: React.FC<PanelCommonProps> = ({
  projectName,
  commonPadding,
  currentTheme,
  language,
}) => {
  const { t } = useTranslation();
  const height = useMemo(() => {
    return (
      rowHeight * thirdLineSize[0].h +
      commonPadding * (thirdLineSize[0].h - 1) -
      80
    );
  }, [commonPadding]);

  const [data, setData] = useState<ColumnConfig['data']>([
    {
      type: 'MySQL',
      value: 0,
    },
  ]);

  const { loading, errorMessage } = usePanelCommonRequest(
    () => statistic.statisticAuditPlanV1({ project_name: projectName }),
    {
      onSuccess: (res) => {
        const dbTypeAuditPlans = res.data.data ?? [];
        if (dbTypeAuditPlans.length > 0) {
          setData(
            dbTypeAuditPlans.reduce<
              Array<{
                type: string;
                value: number;
                category: string;
              }>
            >((acc, cur) => {
              const currentTypes = (cur.data ?? []).map((v) => ({
                type: cur.db_type ?? '',
                value: v.audit_plan_count ?? 0,
                category: v.audit_plan_desc ?? '',
              }));

              return [...acc, ...currentTypes];
            }, [])
          );
        }
      },
    }
  );

  return (
    <PanelWrapper
      loading={loading}
      error={
        errorMessage ? (
          <Result
            status="error"
            title={t('common.request.noticeFailTitle')}
            subTitle={errorMessage}
          />
        ) : null
      }
      title={
        <Space size={12}>
          <Typography.Text>
            {t('projectManage.projectOverview.auditPlanClassification.title')}
          </Typography.Text>

          <Link
            to={`project/${projectName}/auditPlan/create`}
            className="font-size-small"
          >
            {t('auditPlan.action.create')}
          </Link>
        </Space>
      }
    >
      <Column
        {...{ ...config, data }}
        theme={currentTheme}
        locale={language}
        height={height}
      />
    </PanelWrapper>
  );
};

export default AuditPlanClassification;
