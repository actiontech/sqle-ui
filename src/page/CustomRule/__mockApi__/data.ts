import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
