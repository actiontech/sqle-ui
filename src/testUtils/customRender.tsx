import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dictionary } from '../types/common.type';
import { storeFactory } from './mockRedux';
import lightTheme from '../theme/light';
import { ThemeProvider } from '@material-ui/styles';
import { mount, shallow } from 'enzyme';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option]: [...RenderParams]
) => {
  return render(
    <MemoryRouter>
      <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
    </MemoryRouter>,
    option
  );
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};

export const shallowWithTheme = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};
