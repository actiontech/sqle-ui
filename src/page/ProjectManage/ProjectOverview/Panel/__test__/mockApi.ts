import { WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum } from '../../../../../api/common.enum';
import statistic from '../../../../../api/statistic';
import workflow from '../../../../../api/workflow';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetProjectScore = () => {
  const spy = jest.spyOn(statistic, 'GetProjectScoreV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      score: 85,
    })
  );
  return spy;
};

export const mockStatisticsAuditedSQL = () => {
  const spy = jest.spyOn(statistic, 'statisticsAuditedSQLV1');
  spy.mockImplementation(() =>
    resolveThreeSecond(
      {
        risk_sql_count: 420,
        total_sql_count: 1200,
      },
      { otherData: { risk_rate: 35 } }
    )
  );

  return spy;
};

export const mockGetInstanceHealth = () => {
  const spy = jest.spyOn(statistic, 'GetInstanceHealthV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        db_type: 'MySQL',
        health_instance_names: ['mysql-1', 'mysql-2', 'mysql-3'],
        unhealth_instance_names: ['mysql-4', 'mysql-5', 'mysql-6', 'mysql-7'],
      },
      {
        db_type: 'Oracle',
        health_instance_names: ['oracle-1', 'oracle-2', 'oracle-3'],
        unhealth_instance_names: ['oracle-4', 'oracle-5'],
      },
      {
        db_type: 'DB2',
        health_instance_names: ['db-1', 'db-2', 'db-3', 'db-4'],
        unhealth_instance_names: ['db-5', 'db-6'],
      },
    ])
  );
  return spy;
};

export const mockStatisticWorkflowStatus = () => {
  const spy = jest.spyOn(statistic, 'statisticWorkflowStatusV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      waiting_for_audit_count: 32,
      waiting_for_execution_count: 0,
      execution_success_count: 20,
      executing_failed_count: 11,
      rejected_count: 33,
      closed_count: 10,
    })
  );
  return spy;
};

export const mockStatisticRiskWorkflow = () => {
  const spy = jest.spyOn(statistic, 'statisticRiskWorkflowV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        create_user_name: 'admin',
        update_time: '2022-06-09T08:11:52+08:00',
        workflow_id: '123',
        workflow_name: 'test',
        workflow_status: 'exec_failed',
      },
    ])
  );
  return spy;
};

export const mockStatisticAuditPlan = () => {
  const spy = jest.spyOn(statistic, 'statisticAuditPlanV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        data: [
          {
            audit_plan_count: 1,
            audit_plan_type: 'type1',
            audit_plan_desc: 'desc1',
          },
          {
            audit_plan_count: 3,
            audit_plan_type: 'type2',
            audit_plan_desc: 'desc2',
          },
        ],
        db_type: 'MySQL',
      },
      {
        data: [
          {
            audit_plan_count: 2,
            audit_plan_type: 'type1',
            audit_plan_desc: 'desc1',
          },
          {
            audit_plan_count: 1,
            audit_plan_type: 'type2',
            audit_plan_desc: 'desc2',
          },
        ],
        db_type: 'Oracle',
      },
    ])
  );
  return spy;
};

export const mockGetRiskAuditPlan = () => {
  const spy = jest.spyOn(statistic, 'getRiskAuditPlanV1');

  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        audit_plan_name: 'name',
        audit_plan_report_id: '1',
        audit_plan_report_timestamp: '2022-06-09T08:11:52+08:00',
        risk_sql_count: 33,
        trigger_audit_plan_time: '2022-06-09T08:11:52+08:00',
      },
    ])
  );
  return spy;
};

export const mockGetRoleUserCount = () => {
  const spy = jest.spyOn(statistic, 'getRoleUserCountV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        count: 12,
        role: 'dev',
      },
      {
        count: 22,
        role: 'dba',
      },
    ])
  );
  return spy;
};

export const mockGetWorkflowTemplateV1 = () => {
  const spy = jest.spyOn(workflow, 'getWorkflowTemplateV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      allow_submit_when_less_audit_level:
        WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum.warn,
      desc: 'mock test data',
      instance_name_list: ['test'],
      workflow_template_name: 'test',
      workflow_step_template_list: [
        {
          number: 1,
          type: 'sql_review',
          approved_by_authorized: true,
          execute_by_authorized: false,
          assignee_user_name_list: [],
        },
        {
          number: 2,
          type: 'sql_review',
          approved_by_authorized: true,
          execute_by_authorized: false,
          assignee_user_name_list: [],
        },
        {
          number: 3,
          type: 'sql_execute',
          approved_by_authorized: false,
          execute_by_authorized: false,
          assignee_user_name_list: ['admin'],
        },
      ],
    })
  );

  return spy;
};
