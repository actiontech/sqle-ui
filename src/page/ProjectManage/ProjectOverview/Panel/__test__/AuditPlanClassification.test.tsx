/* eslint-disable no-console */
import { act } from '@testing-library/react';
import { mockStatisticAuditPlan } from './mockApi';
import AuditPlanClassification from '../AuditPlanClassification';
import { panelCommonProps } from './index.data';
import { renderWithRouterAndRedux } from '../../../../../testUtils/customRender';
import { resolveErrorThreeSecond } from '../../../../../testUtils/mockRequest';
import { getHrefByText } from '../../../../../testUtils/customQuery';

describe('test AuditPlanClassification', () => {
  let statisticAuditPlanSpy: jest.SpyInstance;
  beforeEach(() => {
    statisticAuditPlanSpy = mockStatisticAuditPlan();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    jest.useRealTimers();
  });
  const error = console.error;

  beforeAll(() => {
    console.error = jest.fn((message: any) => {
      if (message.includes('React does not recognize the')) {
        return;
      }
      error(message);
    });
  });

  afterAll(() => {
    console.error = error;
  });
  test('should match snapshot', async () => {
    const { container } = renderWithRouterAndRedux(
      <AuditPlanClassification {...panelCommonProps} />,
      undefined,
      {
        projectManage: {
          overviewRefreshFlag: false,
        },
      }
    );
    expect(statisticAuditPlanSpy).toBeCalledTimes(1);
    expect(statisticAuditPlanSpy).toBeCalledWith({
      project_name: panelCommonProps.projectName,
    });
    expect(container).toMatchSnapshot();
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getHrefByText('auditPlan.action.create')).toBe(
      `/project/${panelCommonProps.projectName}/auditPlan/create`
    );
    expect(container).toMatchSnapshot();
  });

  test('should render error message', async () => {
    statisticAuditPlanSpy.mockImplementation(() =>
      resolveErrorThreeSecond({ message: 'error message' })
    );

    const { container } = renderWithRouterAndRedux(
      <AuditPlanClassification {...panelCommonProps} />,
      undefined,
      {
        projectManage: {
          overviewRefreshFlag: false,
        },
      }
    );
    await act(async () => jest.advanceTimersByTime(3000));
    expect(container).toMatchSnapshot();
  });
});
