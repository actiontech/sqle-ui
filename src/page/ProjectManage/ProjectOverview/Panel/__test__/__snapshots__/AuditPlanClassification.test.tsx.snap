// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AuditPlanClassification should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
    style="height: 100%;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 12px;"
            >
              <span
                class="ant-typography"
              >
                projectManage.projectOverview.auditPlanClassification.title
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <a
                class="font-size-small"
                href="/project/default/auditPlan/create"
              >
                auditPlan.action.create
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        color="[\\"#fbd44d\\",\\"#43cb77\\",\\"#42a2ff\\"]"
        data="[{\\"type\\":\\"MySQL\\",\\"value\\":0}]"
        height="352"
        legend="{\\"position\\":\\"bottom\\"}"
        locale="zh-CN"
        maxcolumnwidth="40"
        mincolumnwidth="6"
        seriesfield="category"
        theme="light"
        tooltip="{\\"shared\\":false}"
        xfield="type"
        yaxis="{\\"tickInterval\\":1}"
        yfield="value"
      />
    </div>
  </div>
</div>
`;

exports[`test AuditPlanClassification should match snapshot 2`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
    style="height: 100%;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 12px;"
            >
              <span
                class="ant-typography"
              >
                projectManage.projectOverview.auditPlanClassification.title
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <a
                class="font-size-small"
                href="/project/default/auditPlan/create"
              >
                auditPlan.action.create
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        color="[\\"#fbd44d\\",\\"#43cb77\\",\\"#42a2ff\\"]"
        data="[{\\"type\\":\\"MySQL\\",\\"value\\":1,\\"category\\":\\"desc1\\"},{\\"type\\":\\"MySQL\\",\\"value\\":3,\\"category\\":\\"desc2\\"},{\\"type\\":\\"Oracle\\",\\"value\\":2,\\"category\\":\\"desc1\\"},{\\"type\\":\\"Oracle\\",\\"value\\":1,\\"category\\":\\"desc2\\"}]"
        height="352"
        legend="{\\"position\\":\\"bottom\\"}"
        locale="zh-CN"
        maxcolumnwidth="40"
        mincolumnwidth="6"
        seriesfield="category"
        theme="light"
        tooltip="{\\"shared\\":false}"
        xfield="type"
        yaxis="{\\"tickInterval\\":1}"
        yfield="value"
      />
    </div>
  </div>
</div>
`;

exports[`test AuditPlanClassification should render error message 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
    style="height: 100%;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 12px;"
            >
              <span
                class="ant-typography"
              >
                projectManage.projectOverview.auditPlanClassification.title
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <a
                class="font-size-small"
                href="/project/default/auditPlan/create"
              >
                auditPlan.action.create
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-result ant-result-error"
      >
        <div
          class="ant-result-icon"
        >
          <span
            aria-label="close-circle"
            class="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ant-result-title"
        >
          common.request.noticeFailTitle
        </div>
        <div
          class="ant-result-subtitle"
        >
          error
        </div>
      </div>
    </div>
  </div>
</div>
`;
